Composer/packages/server/src/services/project.ts

```diff
@@ -452,36 +451,81 @@ export class BotProjectService {
-
-      if (currentProject !== undefined) {
-        !createFromPva && (await ejectAndMerge(currentProject, jobId));
-        BackgroundProcessManager.updateProcess(jobId, 202, formatMessage('Initializing bot project'));
-        await currentProject.updateBotInfo(name, description, preserveRoot);
-
-        if (schemaUrl && !createFromPva) {
-          await currentProject.saveSchemaToProject(schemaUrl, locationRef.path);
-        }
+      const botsToProcess: { storageId: string; path: string; name: string }[] = [];
 
-        await currentProject.init();
+      // The outcome of our creation might be > 1 bot! We need to determine how many bots we find in this folder.
+      // is this a single bot?
+      if (await StorageService.checkIsBotFolder(newProjRef.storageId, newProjRef.path, user)) {
+        botsToProcess.push({ ...newProjRef, name: name });
+      } else {
+        // or multiple bots?
+        const files = await StorageService.getBlob(newProjRef.storageId, newProjRef.path, user);
+        const childbots = files.children.filter((f) => f.type === 'bot');
+
+        childbots.forEach((b) => {
+          botsToProcess.push({
+            storageId: newProjRef.storageId,
+            path: b.path,
+            name: b.name,
+          });
+        });
+      }
 
+      await Promise.all(
+        botsToProcess.map((botRef) => {
+          // eslint-disable-next-line no-async-promise-executor
+          return new Promise(async (resolve, reject) => {
+            try {
+              log('Open project', botRef);
+              const id = await BotProjectService.openProject(botRef, user);
+
+              // in the case of PVA, we need to update the eTag and alias used by the import mechanism
+              createFromPva && BotProjectService.setProjectLocationData(id, { alias, eTag });
+
+              log('Get Project by Id', id);
+              const currentProject = await BotProjectService.getProjectById(id, user);
+
+              // inject shared content into every new project.  this comes from assets/shared
+              !createFromPva &&
+                (await AssetService.manager.copyBoilerplate(currentProject.dataDir, currentProject.fileStorage));
+
+              if (currentProject !== undefined) {
+                !createFromPva && (await ejectAndMerge(currentProject, jobId));
+                BackgroundProcessManager.updateProcess(jobId, 202, formatMessage('Initializing bot project'));
+
+                log('Updatebot info', id, preserveRoot);
+                await currentProject.updateBotInfo(botRef.name, description, true);
+
+                if (schemaUrl && !createFromPva) {
+                  await currentProject.saveSchemaToProject(schemaUrl, botRef.path);
+                }
+
+                log('Init project', id);
+                await currentProject.init();
+              }
+              resolve(id);
+            } catch (err) {
+              return reject(err);
```

Comment:
How does this error get handled inside Promise.all?

Reply:
the reject should call the promise.all to fail, then the try/catch wrapper will bubble that error up to the client.  I believe this is the desired behavior, but in the future we may want to do something more graceful since the user will need to go to the command line and debug the build failure.